scripts/product.sql

```diff
@@ -287,81 +287,88 @@ ALTER TABLE `catalog_product_bundle_selection`
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_TEXT_ROW_ID_ATTRIBUTE_ID_STORE_ID`,
     ADD CONSTRAINT `CATALOG_PRODUCT_ENTITY_TEXT_ENTITY_ID_ATTRIBUTE_ID_STORE_ID` UNIQUE KEY (`entity_id`,`attribute_id`,`store_id`),
     DROP COLUMN `row_id`;
 
 -- Varchar
 ALTER TABLE `catalog_product_entity_varchar`
     DROP FOREIGN KEY `CAT_PRD_ENTT_VCHR_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_VARCHAR_ENTITY_ID_ATTRIBUTE_ID_STORE_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_VARCHAR_ROW_ID_ATTRIBUTE_ID_STORE_ID`,
     ADD CONSTRAINT `CATALOG_PRODUCT_ENTITY_VARCHAR_ENTITY_ID_ATTRIBUTE_ID_STORE_ID` UNIQUE KEY (`entity_id`,`attribute_id`,`store_id`),
     DROP COLUMN `row_id`;
 
 -- Gallery value to entity
 ALTER TABLE `catalog_product_entity_media_gallery_value_to_entity`
     DROP FOREIGN KEY `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_VAL_ID_ENTT_ID`,
+    DROP INDEX `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
     ADD CONSTRAINT `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_VAL_ID_ENTT_ID` UNIQUE KEY (`value_id`,`entity_id`),
-    DROP COLUMN `row_id`;
+	DROP COLUMN `row_id`;
 
 -- Gallery value
 ALTER TABLE `catalog_product_entity_media_gallery_value`
     DROP FOREIGN KEY `CAT_PRD_ENTT_MDA_GLR_VAL_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_MEDIA_GALLERY_VALUE_ENTITY_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_MEDIA_GALLERY_VALUE_ROW_ID`,
+	DROP INDEX `CAT_PRD_ENTT_MDA_GLR_VAL_ENTT_ID_VAL_ID_STORE_ID`,
     ADD INDEX `CATALOG_PRODUCT_ENTITY_MEDIA_GALLERY_VALUE_ENTITY_ID` (`entity_id`),
     ADD CONSTRAINT `CAT_PRD_ENTT_MDA_GLR_VAL_ENTT_ID_VAL_ID_STORE_ID` UNIQUE KEY (`entity_id`,`value_id`,`store_id`),
-    DROP COLUMN `row_id`;
+	DROP COLUMN `row_id`;
 
 -- Gallery
 ALTER TABLE `catalog_product_entity_gallery`
     DROP FOREIGN KEY `CAT_PRD_ENTT_GLR_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID_ATTRIBUTE_ID_STORE_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ROW_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ROW_ID_ATTRIBUTE_ID_STORE_ID`,
     ADD INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID` (`entity_id`),
     ADD CONSTRAINT `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID_ATTRIBUTE_ID_STORE_ID` UNIQUE KEY (`entity_id`,`attribute_id`,`store_id`),
-    DROP COLUMN `row_id`;
+	DROP COLUMN `row_id`;
 
 -- Tier price
 ALTER TABLE `catalog_product_entity_tier_price`
     DROP FOREIGN KEY `CAT_PRD_ENTT_TIER_PRICE_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `UNQ_E8AB433B9ACB00343ABB312AD2FAB087`,
+    DROP INDEX `UNQ_EBC6A54F44DFA66FA9024CAD97FED6C7`,
     ADD CONSTRAINT `UNQ_E8AB433B9ACB00343ABB312AD2FAB087` UNIQUE KEY (`entity_id`,`all_groups`,`customer_group_id`,`qty`,`website_id`),
-    DROP COLUMN `row_id`;
+	DROP COLUMN `row_id`;
+
+--wishlist 
+ALTER TABLE `wishlist`
+	DROP INDEX `WISHLIST_CUSTOMER_ID`;
```

Comment:
Why this index needs to be removed? Everything else looks good to me :)

Reply:
When I executed php bin/magento setup:upgrade it gave an error on that index. When I deleted that index it suddenly finished the setup:upgrade and everything was working fine.

Reply:
Do you know what was the error? Actually this index is added by the wishlist module shipped with the Open-Source version

Reply:
It said it had a double index WISHLIST_CUSTOMER_ID so it needed to be removed before setup:upgrade could finish.

Reply:
ok, I guess it's related to declarative schema then, could you add a little comment above this line with this short explanation?

Reply:
Just checked again in a my test database of a enterprise version and the wishlist table is not even there. So it is apparently not possible to do before migrating and needs to be manually. I will delete the comment and make a new pull request.